torch/autograd/gradcheck.py

```diff
@@ -314,9 +314,10 @@ def get_jvp_wrt_specific_input(fn, input_idx, inputs, outputs, u, eps) -> List[t
     wrapped_fn = with_prepped_inputs(fn, inputs, input_idx, input_to_perturb, True)
     nbhd_checks_fn = functools.partial(check_outputs_same_dtype_and_shape, eps=eps)
     jvp_fn = get_jvp_fn(wrapped_fn, input_to_perturb, eps, nbhd_checks_fn)
-    if u.layout != torch.sparse_coo:
-        u = u.reshape(input_to_perturb.shape)
-    return compute_numerical_jacobian_cols(jvp_fn, u * eps, input.is_complex())
+    if ur.layout != torch.sparse_coo:
+        ur = ur.reshape(input_to_perturb.shape)
+        ui = ui.reshape(input_to_perturb.shape)
+    return compute_numerical_jacobian_cols(jvp_fn, ur * eps, input.is_complex(), ui * eps)
```

Comment:
nit: ordering of the arguments looks a bit weird here. Maybe use `delta_i=ui * eps` to make it less weird to have the `input.is_complex()` in between the two.

Reply:
I've updated to use only a single delta, see other comment above